ww3tools/ProcSat_interpolation.py

```diff
@@ -0,0 +1,515 @@
+
+    full_model_times = np.full(len(satellite_times), np.nan)
+    full_model_times[mask] = model_times_for_interpolated_values
+
+    return full_interpolated_values, full_model_times
+
+def spatial_mask(satellite_latitudes, satellite_longitudes, model_data):
+    min_latitude = model_data.latitude.min().values.item()
+    max_latitude = model_data.latitude.max().values.item()
+    min_longitude = model_data.longitude.min().values.item()
+    max_longitude = model_data.longitude.max().values.item()
+
+    within_lat_bounds = (satellite_latitudes >= min_latitude) & (satellite_latitudes <= max_latitude)
+    within_lon_bounds = (satellite_longitudes >= min_longitude) & (satellite_longitudes <= max_longitude)
+    return within_lat_bounds & within_lon_bounds
+
+
+
+
+# NETCDF interpolator
+def interpolate_netcdf(model_data_directory, model_data_pattern, satellite_file, output_file, model_name):
+    # Using glob to find all files in the directory that match the pattern
+    model_data_files = glob.glob(model_data_directory + model_data_pattern)
+
+    # Check if any files are found
+    if not model_data_files:
+        raise ValueError("No files found in the specified directory matching the pattern.")
+
+    # Sort the files to ensure they are in the correct order
+    model_data_files.sort()
+
+    def extract_reference_times(file_path):
+        ds = xr.open_dataset(file_path)
+        reference_time = ds.time.attrs['reference_time']
+        reference_date = ds.time.attrs['reference_date']
+        ds.close()  # Close the dataset after extraction
+        return reference_time, reference_date
+
+    model_reference_time_unix = extract_reference_times(model_data_files[0])[0]  #added
+    reference_time_unix = extract_reference_times(model_data_files[0])[0]  # Returns Unix timestamp as integer or float
+
+    # Load and concatenate the model data files
+    model_data_list = [xr.open_dataset(file) for file in model_data_files]
+    model_data_combined = xr.concat(model_data_list, dim='time')
+    model_data_combined.close()  # Close the files after loading
+
+    # Convert longitudes from -180 to 180 to 0 to 360, if necessary
+    model_longitudes = model_data_combined['longitude'].values
+    if model_longitudes.min() < 0:
+        model_longitudes[model_longitudes < 0] += 360
+        model_data_combined = model_data_combined.assign_coords(longitude=model_longitudes)
+
+    # Convert model time to UNIX time and ensure it's double
+    model_time_seconds = np.array((model_data_combined['time'].values - np.datetime64('1970-01-01T00:00:00')) / np.timedelta64(1, 's')).astype('double')
+    max_model_time = model_time_seconds.max()
+
+    # Defining model_times correctly for use in interpolation
+    model_times = model_time_seconds
+
+    # Load satellite data
+    satellite_data = xr.open_dataset(satellite_file)
+    satellite_longitudes = satellite_data['longitude'].values
+    satellite_longitudes[satellite_longitudes < 0] += 360
```

Comment:
I do not see this check for grib2, should this be added to the grib2 section?

Reply:
It is added.

Reply:
@Ghazal-Mohammadpour I'm having trouble finding where this check was added for the satellite_longitudes in the grib2 section, would it be possible to point me to a line number?

Reply:
This is now on line 172. Thanks for adding this @Ghazal-Mohammadpour